### PR TITLE
[CdapIO] CdapIO and SparkReceiverIO updates

### PR DESCRIPTION
**Pull request by Amar3tto**
- Added `pullFrequencySec` parameter - delay in seconds between polling for new records updates.
- Added `startOffset` parameter - inclusive start offset from which the reading should be started.
- Refactoring (`Plugin`, `MappingUtils`, context classes)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AKosolapov** [comment]
Run Java PreCommit

**Comment by AKosolapov** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by Lizzfox** [comment]
Run Java PreCommit

**Comment by Lizzfox** [comment]
Run Java_Examples_Dataflow_Java17 PreCommit

**Comment by Amar3tto** [comment]
Run Java PreCommit

**Comment by AKosolapov** [comment]
Run Java PreCommit

**Comment by Lizzfox** [comment]
R: @aromanenko-dev 
R: @johnjcasey 
R: @chamikaramj 

This is the PR with CDAP and SparkReceiver IOs' changes (related to [the comment](https://github.com/apache/beam/pull/23501#pullrequestreview-1199788683) from the https://github.com/apache/beam/pull/23501 PR). 

Thanks!

File: sdks/java/io/cdap/src/main/java/org/apache/beam/sdk/io/cdap/context/FailureCollectorWrapper.java [hunk 1]
```diff
@@ -41,6 +44,21 @@ public ValidationFailure addFailure(String message, @Nullable String correctiveA
 
   @Override
   public ValidationException getOrThrowException() throws ValidationException {
+
+    // We skip schema field validation errors because they are CDAP oriented and don't affect
+    // anything in our case
+    for (Iterator<ValidationFailure> failureIterator = failuresCollection.iterator();
```
[the remark this answers is not in the document]
Comment (Amar3tto) [reply sdks/java/io/cdap/src/main/java/org/apache/beam/sdk/io/cdap/context/FailureCollectorWrapper.java]: Done

**Comment by Lizzfox** [comment]
@johnjcasey 
All comments have been addressed, thanks!

**Review by johnjcasey — APPROVED** [review]

**Pull request merged**